packages/commons/src/auth/jwk.ts

```diff
@@ -45,17 +46,27 @@ function assertNotPrivateKey(key: string): void {
   throw notAllowedPrivateKeyException();
 }
 
+export function assertValidRSAKeyLength(
+  keyString: string,
+  keyobject: KeyObject,
+  minLength: number = 2048
+): void {
+  const length = keyobject.asymmetricKeyDetails?.modulusLength;
+  if (!length || keyobject.asymmetricKeyType !== "rsa") {
+    throw invalidKey(keyString, "Not an RSA key");
+  }
+  if (length < minLength) {
+    throw invalidKeyLength(length, minLength);
+  }
+}
```

Comment:
I would split this into two validators to keep things cleaner ⬇️ 
Also, I would suggest **not** using `invalidKey` because looking at the function params, it requires a `kid`, but we are passing the entire key string. I would not print the entire key string, let's just create an error called something like `notAnRSAKey`, without parameters, that just prints a message like `'Provided key is not an RSA key.'`. For example, we do something similar inside `assertNotPrivateKey` or `assertNotCertificate`.

If `invalidKey` is then unused, we can remove it!

```suggestion
export function assertValidRSAKey(
  key: KeyObject
): void {
  if (key.asymmetricKeyType !== 'rsa') {
    throw notAnRSAKey();
  }
}

export function assertValidRSAKeyLength(
  key: KeyObject,
  minLength: number = 2048
): void {
  const length = key.asymmetricKeyDetails?.modulusLength;
  if (!length || length < minLength) {
    throw invalidKeyLength(length, minLength);
  }
}